### PR TITLE
Refactor GistTag

### PR DESCRIPTION
**Pull request by ashmaroli**
- `GistTag#context_contains_key?` uses both `:has_key?` and `key?`. So its effectively a stalemate.
- `GistTag.client` is a `public_class_method`. It should be placed above the `private` access modifier
https://github.com/jekyll/jekyll-gist/blob/1ca4a4dd73798d6d9596089b9d65e88e9f778b5c/lib/jekyll-gist/gist_tag.rb#L97

### CODE REVIEW

File: lib/jekyll-gist/gist_tag.rb [hunk 1]
```diff
@@ -35,20 +35,26 @@ def render(context)
...
+      end
+
       private
 
       def determine_arguments(input)
         matched = input.match(%r!\A([\S]+|.*(?=\/).+)\s?(\S*)\Z!)
         [matched[1].strip, matched[2].strip] if matched && matched.length >= 3
       end
 
+      # rubocop:disable Style/PreferredHashMethods
```
Comment (DirtyF) [inline lib/jekyll-gist/gist_tag.rb]: I'd rather avoid comments in files, it defeats the purpose of having an automatically generated todo

File: lib/jekyll-gist/gist_tag.rb [hunk 2]
```diff
@@ -47,6 +47,7 @@ def determine_arguments(input)
       end
 
       # rubocop:disable Style/PreferredHashMethods
+      # Remove the check for `:has_key?` when the plugin drops support for `< Liquid 4.0`
```
Comment (DirtyF) [inline lib/jekyll-gist/gist_tag.rb]: As https://pages.github.com/versions/  supports it, I'm fine with requiring Liquid 4.0.
That's up to someone from GitHub to decide.
/cc  @parkr

Comment (parkr) [reply lib/jekyll-gist/gist_tag.rb]: Yeah, requiring Liquid 4 is fine by me, but give a major bump to this gem if you do that. For now, backwards compatibility is good.

**Review by DirtyF — APPROVED** [review]
LGTM.

**Comment by ashmaroli** [comment]
@mattr-, @parkr I'd like to move the following **regex** to a `private_constant`. What would be a suitable name for the constant?
https://github.com/jekyll/jekyll-gist/blob/c74391b6d8bae1897fc1cb70234eb5e4119d8a62/lib/jekyll-gist/gist_tag.rb#L49

**Comment by mattr-** [comment]
I wouldn't bother moving it to a constant at all. Very little gain for that change when it's already in a method that describes what's going on. Moving the regexp to a constant makes it more important in the class than it is. That's not the class' main concern, so best to leave it hidden inside that method, IMO.

**Comment by mattr-** [comment]
For a PR that was only supposed to be about rubocop offenses, this is starting to get off track. Let's cap it off here and then do further work in other PRs if necessary.

**Comment by ashmaroli** [comment]
@mattr-  It had stopped being about RuboCop offenses much before I renamed the PR title. But I agree, its time to defer any more changes.

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev

**Pull request merged**